tests/components/konnected/test_panel.py

```diff
@@ -111,87 +109,96 @@ def mock_constructor(host, port, websession):
         "mac": "11:22:33:44:55:66",
         "settings": {},
     }
-    device = panel.AlarmPanel(hass, entry)
-    await device.async_save_data()
-    await device.async_connect()
+
+    # setup the integration and inspect panel behavior
+    assert (
+        await async_setup_component(
+            hass,
+            panel.DOMAIN,
+            {
+                panel.DOMAIN: {
+                    panel.CONF_ACCESS_TOKEN: "arandomstringvalue",
+                    panel.CONF_API_HOST: "http://192.168.1.1:8123",
+                }
+            },
+        )
+        is True
+    )
+
+    # confirm panel instance was created and configured
+    # hass.data is the only mechanism to get a reference to the created panel instance
+    device = hass.data[panel.DOMAIN][panel.CONF_DEVICES]["112233445566"]["panel"]
     await device.update_switch("1", 0)
 
     # confirm the correct api is used
     # pylint: disable=no-member
-    assert device.client.put_device.call_count == 1
-    assert device.client.put_zone.call_count == 0
+    assert mock_panel.put_device.call_count == 1
+    assert mock_panel.put_zone.call_count == 0
 
     # confirm the settings are sent to the panel
     # pylint: disable=no-member
-    assert device.client.put_settings.call_args_list[0][1] == {
+    assert mock_panel.put_settings.call_args_list[0][1] == {
         "sensors": [{"pin": "1"}, {"pin": "2"}, {"pin": "5"}],
         "actuators": [{"trigger": 0, "pin": "8"}, {"trigger": 1, "pin": "9"}],
         "dht_sensors": [{"poll_interval": 3, "pin": "6"}],
         "ds18b20_sensors": [{"pin": "7"}],
         "auth_token": "11223344556677889900",
         "blink": True,
         "discovery": True,
-        "endpoint": "192.168.1.1/api/konnected",
+        "endpoint": "http://192.168.1.1:8123/api/konnected",
     }
 
     # confirm the device settings are saved in hass.data
-    assert hass.data[panel.DOMAIN][panel.CONF_DEVICES] == {
-        "112233445566": {
-            "binary_sensors": {
-                "1": {
-                    "inverse": False,
-                    "name": "Konnected 445566 Zone 1",
-                    "state": None,
-                    "type": "door",
-                },
-                "2": {
-                    "inverse": True,
-                    "name": "winder",
-                    "state": None,
-                    "type": "window",
-                },
-                "3": {
-                    "inverse": False,
-                    "name": "Konnected 445566 Zone 3",
-                    "state": None,
-                    "type": "door",
-                },
+    assert device.stored_configuration == {
```

Comment:
Instead we should assert that `mock_panel.put_settings` is called with the correct configuration during device sync.